includes/class-amp-story-post-type.php

```diff
@@ -936,6 +936,31 @@ public static function enqueue_block_editor_scripts() {
 			);
 		}
 
+		/**
+		 * Filter list of allowed video mime types.
+		 *
+		 * This can be used to add additionally supported formats, for example by plugins
+		 * that do video transcoding.
+		 *
+		 * @since 1.3
+		 *
+		 * @param array Allowed video mime types.
+		 */
+		$allowed_video_mime_types = apply_filters( 'amp_story_allowed_video_types', [ 'video/mp4' ] );
```

Comment:
There is another filter in core that is very similar `wp_video_extensions` could someone filtering that value effect anything?

Reply:
Should the values in this array be compared to values in the array returned by `wp_get_mime_types()`

Reply:
> There is another filter in core that is very similar `wp_video_extensions` could someone filtering that value effect anything?

Story is using static mime type `video/mp4` right now, `wp_video_extensions` only adds extensions so it should not cause any affect here.

> Should the values in this array be compared to values in the array returned by wp_get_mime_types()

Compared to see if the values received by `amp_story_allowed_video_types` are valid or not?

Reply:
I am worrying if this array should be compared against values found in `wp_get_mime_types()`. What is the point of adding a mime type that core itself doesn't support? So maybe using `array_intersect` of the two arrays?

Reply:
Since `wp_get_mime_types` uses `mime_types` filter internally which can be modified, relying on that won't be full proof either, as the function contains any other file mime types, comparing received data isn't good as it can contain any type passed by services using that filter.

Only full proof way would be checking against static valid video formats list, which would have to be updated in some way.

Reply:
I don't think we need a bulletproof sanity check here when someone's `doing_it_wrong™`. `array_intersect` with `wp_get_mime_types` is good enough IMO.

Reply:
@spacedmonkey I've made the changes as suggested, please check and let me know if it's ok.

I've kept the previous validation to add `video/mp4` by default, if received data is empty or not an array, let me know if that needs change as well. Thanks.